eo-maven-plugin/src/main/java/org/eolang/maven/SafeMojo.java

```diff
@@ -285,4 +300,41 @@ private static void closeTojos(final Closeable res) throws MojoFailureException
         }
     }
 
+    /**
+     * Interrupts thread if time limit is reached.
+     *
+     * @param thread Thread to interrupt.
+     * @param sec Time limit in seconds.
+     */
+    private static void waitAndInterrupt(final Thread thread, final int sec) {
+        new Thread(
+            () -> {
+                try {
+                    Thread.sleep(TimeUnit.SECONDS.toMillis(sec));
+                    synchronized (thread) {
```

Comment:
@volodya-lombrozo since you don't check for `isAlive()`, synchronization is redundant, I believe